### PR TITLE
[SYCL][DPCPP] Fix group_broadcast test for Boolean data type

### PR DESCRIPTION
**Pull request by uditagarwal97**
Currently, the group_broadcast test fails for boolean variables.

### CODE REVIEW

File: tests/group_functions/group_broadcast.h [hunk 1]
```diff
@@ -195,8 +195,12 @@ void broadcast_sub_group(sycl::queue& queue) {
           if (sub_group.get_local_linear_id() == 0) res_acc[3] = local_var;
 
           // return the sub-group size
```
Comment (keryell) [inline tests/group_functions/group_broadcast.h:197]:
```suggestion
          // Return the sub-group size when possible or just its parity
```

**Review by keryell — CHANGES_REQUESTED** [review]

File: tests/group_functions/group_broadcast.h [hunk 2]
```diff
@@ -195,8 +195,12 @@ void broadcast_sub_group(sycl::queue& queue) {
...
-          if (sub_group.get_local_linear_id() == 0)
-            res_acc[4] = sub_group.get_local_linear_range();
+          if (sub_group.get_local_linear_id() == 0){
```
Comment (bader) [inline tests/group_functions/group_broadcast.h:198]:
```suggestion
          if (sub_group.get_local_linear_id() == 0) {
```

**Comment by tomdeakin** [comment]
Looks like changes still in progress. Please finish, and review.

**Review by keryell — APPROVED** [review]
Thanks!

**Pull request merged**